File: MovieTicketBooking/Movie Booking Java/src/main/java/org/example/controllers/MovieController.java
package org.example.controllers;

import lombok.AllArgsConstructor;
import lombok.NonNull;
import org.example.model.Movie;
import org.example.service.MovieService;

@AllArgsConstructor
public class MovieController {

    private final MovieService movieService;

    public Movie addMovie(@NonNull final String name) {

        return movieService.addMovie(name);
    }

    public Movie getMovie(@NonNull final String id) {

        return movieService.getMovie(id);
    }
}

File: MovieTicketBooking/Movie Booking C#/Controllers/MovieController.cs
using MovieTicketBooking.Services;

namespace MovieTicketBooking.Controllers
{
    public class MovieController
    {
        MovieService _movieService;
        public MovieController(MovieService movieService)
        {
            _movieService = movieService;
        }
        public string Add(string name)
        {
            return _movieService.AddMovie(name);
        }
    }
}